### PR TITLE
Worker connection on game thread fix

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This was a mistake we made when adding the option to run the spatial worker connection on the game thread. 

#### Primary reviewers
@samiwh  @aleximprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 1]
```diff
@@ -784,6 +784,10 @@ ERPCResult USpatialSender::SendRPCInternal(UObject* TargetObject, UFunction* Fun
 
 			FWorkerComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Payload, ComponentId, RPCInfo.Index);
 
+#if TRACE_LIB_ACTIVE
+			//ComponentUpdate.Trace = Payload.Trace;
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Looks like these lines were pushed by mistake?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 2]
```diff
@@ -784,6 +784,10 @@ ERPCResult USpatialSender::SendRPCInternal(UObject* TargetObject, UFunction* Fun
 
 			FWorkerComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Payload, ComponentId, RPCInfo.Index);
 
+#if TRACE_LIB_ACTIVE
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Believe this fix made it in another PR

**Comment by MatthewSandfordImprobable** [comment]
> Change looks good - is this just a semantical change, or were there performance/functional improvements? Just because doing this late Dispatch vs early Flush seems much the same in terms of timings.

@samiwh

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Think Sami removed that PR

**Comment by MatthewSandfordImprobable** [comment]
Discussed issue with Michael offline

**Pull request merged**